exporter/clickhouseexporter/factory.go

```diff
@@ -102,3 +104,26 @@ func createTracesExporter(
 		exporterhelper.WithRetry(c.RetrySettings),
 	)
 }
+
+func createMetricExporter(
+	ctx context.Context,
+	set exporter.CreateSettings,
+	cfg component.Config,
+) (exporter.Metrics, error) {
+	c := cfg.(*Config)
+	exporter, err := newMetricsExporter(set.Logger, c)
+	if err != nil {
+		return nil, fmt.Errorf("cannot configure clickhouse metrics exporter: %w", err)
+	}
+
+	return exporterhelper.NewMetricsExporter(
+		ctx,
+		set,
+		cfg,
+		exporter.pushMetricsData,
+		exporterhelper.WithShutdown(exporter.Shutdown),
```

Comment:
```suggestion
		exporterhelper.WithShutdown(exporter.shutdown),
```
You can probably make shutdown internal